feature/a1

ArtyomTsirsis opened the pull request.

ArtyomTsirsis committed:
Main class/small changes

ArtyomTsirsis committed:
Merge branch 'develop' into feature/a1

ArtyomTsirsis committed:
Main class/small changes

ArtyomTsirsis committed:
HeroDTO/Responses

ArtyomTsirsis committed:
Repo services/package refactor

ArtyomTsirsis committed:
one more refactor

ArtyomTsirsis committed:
naming fix and test classes created

nextlevelplay commented on src/main/java/com/game/repository/hero/HeroRepository.java:
```diff
@@ -1,28 +1,28 @@
 import java.util.Map;
 
+@Repository
 public class HeroRepository {
```
Сразу не заметил, у тебя HeroRepository - это класс не имплементирующий интерфейс - что приводит к жёсткой связке в наших сервисах! Надо создать интрефейс и  Именно его предавать в сервисы. Когда подключишь настоящую базу, всё что тебе нужно будет сделать - это снять с этого репозитория аннотацию и поставить её на настоящую.

nextlevelplay replied:
Типо такого RealDataBase implement Repository (где Репозиторий - это интерфейс), иду в сервисы, чтобы развить мысль...

nextlevelplay commented on src/main/java/com/game/repository/services/CreateHeroService.java:
```diff
@@ -0,0 +1,32 @@
+public class CreateHeroService {
+
+    @Autowired
+    private HeroRepository repository;
```
Именно про это я и говорил - в игре пропадёт HeroRepository - а сервис будет настырно искать его...
Но если ты сделаешь вот так:
@Autowired
private Repository repository;   Сервису абсолютно будет наплевать какая реализация к нему пришла - на массиве или настоящая база. В твоём случае он будет искать на массиве - так как она жёстко привязана.
и чтобы это исправить придётся переколбасить все сервисы в ручную. Главное, чтобы небыло две имплементации и с анотациями в одно время, так как они превратятся в Beans и Spring выдаст ошибку, что нащёл сразу два файла с реализацией репозитория(Ну или можно как ты навесить @Qualifier - но это опять лишний код).

nextlevelplay replied:
Сорян, наверника - это можно было пофиксить раньше, но я просто мерджил не вникая в код.

ArtyomTsirsis committed:
Too much refactor, too much changes, sorry. )
Added repo, changed skills, Database connections added. Script for our tables in mySQL(commands there too like Ruslan made)

ArtyomTsirsis committed:
Too much refactor, too much changes, sorry. )
Added repo, changed skills, Database connections added. Script for our tables in mySQL(commands there too like Ruslan made), battle commented

ArtyomTsirsis commented:
Repo interface сделал.
